### PR TITLE
[OneExplorer] Validate input for rename

### PR DESCRIPTION
**Pull request by dayo09**
This commit validates an input for renaming.
It checks (1) ext, (2) file already exists.

ONE-vscode-DCO-1.0-Signed-off-by: Dayoung Lee <dayoung.lee@samsung.com>

### CODE REVIEW

**Comment by dayo09** [comment]
@yunjayh 
The feature from [this conversation](https://github.com/Samsung/ONE-vscode/pull/674#discussion_r882313698) :-D

> 
> How about giving proper extension name if it's not given by user? I cannot imagine the danger if we apply following change.
> 
> @[dayo09](https://github.com/dayo09) dayo09 [4 hours ago](https://github.com/Samsung/ONE-vscode/pull/674#discussion_r882321163) • 
> IMO, it's a little bit too much... Maybe it will bring about much confusion to user. Well, how about just putting an error message instead in case of user puts incorrect ext?
> 
> @[yunjayh](https://github.com/yunjayh) yunjayh [4 hours ago](https://github.com/Samsung/ONE-vscode/pull/674#discussion_r882324952)
> IMO, it's a little bit too much... Maybe it will bring about much confusion to user.
> 
> I agree with you. Since there is a warning message at the bottom of input box, user would pay attention with his/her file naming. IMHO Defining the situation when the error message is popping up will be pretty difficult.
> Then, never mind about the above suggestion!
> 
> @[dayo09](https://github.com/dayo09) dayo09 [3 hours ago](https://github.com/Samsung/ONE-vscode/pull/674#discussion_r882327319)
> @yunjayh I think we can use 'validateInput' API to check if user has given a right ext. I will apply that on the next pr. thanks for your opinion.

**Comment by dayo09** [comment]
[Here](https://github.sec.samsung.net/RS7-RuntimeNTools/secret/issues/187#issuecomment-1523764) is a gif link. PTAL.

File: src/OneExplorer/OneExplorer.ts [hunk 1]
```diff
@@ -185,11 +185,26 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
...
+      const oldpath = oneNode.node.path;
+      const dirpath = path.dirname(oneNode.node.uri.fsPath);
+      const newpath: string = `${dirpath}/${newname}`;
+
+      if (path.extname(newpath) !== path.extname(oldpath)) {
+        return `File ext must be (${path.extname(oldpath)})`;
+      }
+
+      if (fs.existsSync(newpath)) {
+        return `Invalid: Path already exists!`;
```
Comment (yunjayh) [inline src/OneExplorer/OneExplorer.ts:198]:
```suggestion
        return `Invalid: File already exists!`;
```

It looks not to renaming directory. So, File already exists would be more clear for user IMO

File: src/OneExplorer/OneExplorer.ts [hunk 2]
```diff
@@ -185,11 +185,26 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
       warningMessage = 'WARNING: Renaming may result in some unexpected changes on the tree view.';
     }
 
+    const validateInputPath = (newname: string): string|undefined => {
+      const oldpath = oneNode.node.path;
+      const dirpath = path.dirname(oneNode.node.uri.fsPath);
+      const newpath: string = `${dirpath}/${newname}`;
```
Comment (hyunsik-yoon) [inline src/OneExplorer/OneExplorer.ts:191]: (optional) how about using `path.join` for future when we support Windows? 😸 

```suggestion
      const newpath: string = path.join(${dirpath}, ${newname});
```

File: src/OneExplorer/OneExplorer.ts [hunk 3]
```diff
@@ -185,11 +185,26 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
...
+      const oldpath = oneNode.node.path;
+      const dirpath = path.dirname(oneNode.node.uri.fsPath);
+      const newpath: string = path.join(dirpath, newname);
+
+      if (path.extname(newpath) !== path.extname(oldpath)) {
+        return `File ext must be (${path.extname(oldpath)})`;
+      }
+
+      if (fs.existsSync(newpath)) {
+        return `Invalid: File already exists!`;
```
Comment (dayo09) [inline src/OneExplorer/OneExplorer.ts:198]:
```suggestion
        return `A file or folder ${newpath} already exists at this location. Please choose a different name.`;
```

Comment (dayo09) [reply src/OneExplorer/OneExplorer.ts]: It's what the vscode default explorer says. Let's sync with theirs.

**Review by hyunsik-yoon — APPROVED** [review]

**Review by yunjayh — APPROVED** [review]

**Review by hyunsik-yoon — APPROVED** [review]

**Pull request merged**